src/Component.php

```diff
@@ -2,15 +2,66 @@
 
 declare(strict_types=1);
 
-namespace MyComponent;
+namespace Keboola\ExTeradata;
 
+use Dibi\Connection;
 use Keboola\Component\BaseComponent;
+use Keboola\Component\UserException;
 
 class Component extends BaseComponent
 {
+    private function createConnection(string $host, string $usename, string $password): Connection
+    {
+        return new Connection([
+            'dsn' => sprintf('DRIVER={Teradata};DBCName=%s', $host),
+            'driver'   => 'odbc',
+            'username' => $usename,
+            'password' => $password,
+        ]);
+    }
+
     public function run(): void
     {
-        // @TODO implement
+        $parameters = $this->getConfig()->getParameters();
+
+        $credentials = $parameters['db'] ?? null;
+        if ($credentials === null) {
+            throw new UserException('Database credentials must be set.');
```

Comment:
Není tohle ošetřené už na úrovni confi definition? A nešlo by tam ošetřit i https://github.com/keboola/ex-teradata/pull/1/files#diff-c694a29ec41709ca1a5f5e8a29f6467eR32 ? 
Myslím že je to dost podobné jako u DWH manageru https://github.com/keboola/app-snowflake-dwh-manager/blob/master/src/ConfigDefinition.php